src/runtime.h

```diff
@@ -2502,6 +2434,71 @@ static STRING ___show_LIST(LIST l)
 	return gc_strdup(show_buffer);
 }
 
+
+
+static regex_t* memoized_regcomp(STRING reg_str)
+{
+	regex_t* reg;
+	if (___has(box_STRING(reg_str), memoized_regexes)) reg = ___D(box_STRING(reg_str), memoized_regexes).ptr;
+	else
+	{
+		reg = gc_flatmalloc(sizeof *reg);
+		const int status = regcomp(reg, reg_str, REG_EXTENDED | REG_NEWLINE);
```

Comment:
Note that previously Regex also used the REG_NOSUB flag when compiling (Regex-match does not since it needs those subexpressions). I have no idea if it could affect the performance for Regex,  though.

Reply:
Yeah I considered having different memoization buffers for the two functions but this felt easier - if the performance becomes  concern I'll do that.